modules/service/src/main/scala/lucuma/odb/graphql/mapping/QueryMapping.scala

```diff
@@ -140,18 +142,31 @@ trait QueryMapping[F[_]] extends Predicates[F] {
         OrderBy(OrderSelections(List(OrderSelection[Short](ProposalStatusMetaType / "ordinal"))), child)
       }
 
+  private def datasetPredicate(
+    rDid: Result[Option[model.sequence.Dataset.Id]],
+    rRef: Result[Option[DatasetReference]],
+    pred: DatasetPredicates
+  ): Result[Predicate] =
+    (rDid, rRef).parTupled.map { (did, ref) =>
+      and(List(
+        did.map(pred.id.eql).toList,
+        ref.map(r => pred.referenceLabel.eql(r)).toList
+      ).flatten) match {
+        case True => False // neither did nor ref was supplied
```

Comment:
I like this. 😆

Reply:
Yeah Grackle collapses an empty list down to `Predicate.True` but in this case it shouldn't match anything.